index.html

```diff
@@ -77,67 +78,76 @@ <h2>Examples of Usage</h2>
+          <li>If <i>isInputPendingOptionsInit.includeContinuous</i> is set and is not null, then set the value of <i>options.includeContinuous</i> to the value of <i>isInputPendingOptionsInit.includeContinuous</i>. Otherwise, set <i>isInputPendingOptionsInit.includeContinuous</i> to <i>false</i>.</li>
+          <li>Return <i>options</i>.</li>
+        </ol>
+        <p class="note"><a data-link-for="Scheduling">isInputPending</a> is expected to be called many times within a single frame and so the API is designed to be as low overhead as possible. Many browsers have optimizations that can be applied if the options object is instantiated once and is of a specific type verses having to check a generic object each time; thus in order to allow for browser optimization, <a>IsInputPendingOptions</a> objects require instantiation.</p>
+      </section>
     </section>
-    <section data-link-for="Scheduling" data-dfn-for="Scheduling">
+    <section data-dfn-for="Scheduling">
       <h2>The <dfn>Scheduling</dfn> interface</h2>
       <pre class='idl'>
-        [<dfn data-cite="webidl#idl-Exposed">Exposed</dfn>=<dfn data-cite="HTML5#the-window-object">Window</dfn>] interface Scheduling {
-           <dfn data-cite="webidl#idl-boolean">boolean</dfn> isInputPending(optional sequence&lt;InputType&gt; inputTypes = []);
-           boolean canCheckInputPending(optional InputType inputType = "");
+        [<dfn data-cite="webidl#idl-Exposed">Exposed</dfn>=Window] interface Scheduling {
+           boolean isInputPending(optional IsInputPendingOptions isInputPendingOptions = null);
         };
       </pre>
       <section>
-        <h3>The <dfn data-link-for="isInputPending">isInputPending</dfn> Method</h3>
-        <p>The <code>isInputPending</code> method returns a value based on the strings listed in <code>inputTypes</code>. If the <code>isInputPending</code> method is called then the user agent MUST run the following steps:</p>
+        <h3>The <dfn>isInputPending</dfn> Method</h3>
+        <p>The <code>isInputPending</code> method returns a value based on the options set listed in <code>isInputPendingOptions</code>. If the <code>isInputPending</code> method is called then the user agent MUST run the following steps:</p>
         <ol>
-          <li>If <code>inputTypes</code> is empty then let <code>inputTypes</code> be an array of <code>SupportedEventTypes</code>.</li>
-          <li>Let <code>pendingResult</code> be false.</li>
-          <li>The user agent MAY at this step set <code>pendingResult</code> to true for any reason.</li>
+          <li>Let <i>pendingResult</i> be false.</li>
+          <li>If <i>isInputPendingOptions</i> is not set then set <i>isInputPendingOptions</i> to be the output from running the <a>IsInputPendingOptions</a> constructor with no input options.</li>
+          <li>The user agent MAY at this step set <i>pendingResult</i> to true for any reason.</li>
```

Comment:
Why is this line there? Doesn't it violate the security note at the end of the document?

Reply:
You're right, based on the updated note we should remove it.